app/src/sharedTest/java/org/oppia/android/app/testing/NavigationDrawerTestActivityTest.kt

```diff
@@ -412,160 +337,102 @@ class NavigationDrawerTestActivityTest {
+        .inRoot(isDialog())
+        .check(matches(isDisplayed()))
+      onView(withText(R.string.home_activity_back_dialog_exit))
+        .inRoot(isDialog())
+        .perform(click())
       intended(hasComponent(ProfileChooserActivity::class.java.name))
     }
   }
 
+  @RunOn(TestPlatform.ESPRESSO)
   @Test
-  // TODO(#973): Fix NavigationDrawerTestActivityTest
-  @Ignore
-  fun testNavigationDrawerTestActivity_openNavigationDrawer_selectSwitchProfileMenu_showsExitToProfileChooserDialog_clickCancel_checkDrawerIsClosed() { // ktlint-disable max-line-length
+  fun testNavigationDrawerTestActivity_selectSwitchProfileMenu_clickCancel_checkDrawerIsClosed() {
     launch(NavigationDrawerTestActivity::class.java).use {
-      onView(withId(R.id.home_activity_drawer_layout)).perform(open())
+      it.openNavigationDrawer()
       onView(withText(R.string.menu_switch_profile)).perform(click())
-      onView(withText(R.string.home_activity_back_dialog_message)).check(matches(isDisplayed()))
-      onView(withText(R.string.home_activity_back_dialog_cancel)).perform(click())
+      onView(withText(R.string.home_activity_back_dialog_message))
+        .inRoot(isDialog())
+        .check(matches(isDisplayed()))
+      onView(withText(R.string.home_activity_back_dialog_cancel))
+        .inRoot(isDialog())
+        .perform(click())
+      testCoroutineDispatchers.runCurrent()
       onView(withId(R.id.home_activity_drawer_layout)).check(matches(isClosed()))
-      onView(withId(R.id.home_activity_drawer_layout)).perform(open())
-      onView(
-        allOf(
-          instanceOf(TextView::class.java),
-          withParent(withId(R.id.home_activity_toolbar))
-        )
-      ).check(matches(withText(R.string.menu_home)))
     }
   }
 
   @Test
-  // TODO(#973): Fix NavigationDrawerTestActivityTest
-  @Ignore
   fun testNavigationDrawerTestActivity_selectSwitchProfile_orientationChange_checkDialogVisible() {
     launch(NavigationDrawerTestActivity::class.java).use {
-      onView(withId(R.id.home_activity_drawer_layout)).perform(open())
+      it.openNavigationDrawer()
       onView(withText(R.string.menu_switch_profile)).perform(click())
-      onView(withText(R.string.home_activity_back_dialog_message)).check(matches(isDisplayed()))
+      onView(withText(R.string.home_activity_back_dialog_message))
+        .inRoot(isDialog())
+        .check(matches(isDisplayed()))
       onView(isRoot()).perform(orientationLandscape())
       onView(withText(R.string.home_activity_back_dialog_message)).check(matches(isDisplayed()))
     }
   }
 
   @Test
-  // TODO(#973): Fix NavigationDrawerTestActivityTest
-  @Ignore
-  fun testNavigationDrawerTestActivity_openNavigationDrawer_selectHelpMenuInNavigationDrawer_clickNavigationDrawerHamburger_navigationDrawerIsOpenedAndVerifiedSuccessfully() { // ktlint-disable max-line-length
+  fun testNavigationDrawerTestActivity_openNavDrawer_selectHelpMenu_opensHelpActivity() { // ktlint-disable max-line-length
```

Comment:
Can we remove the ktlint disabled directive now? Prefer removing all of them in this file while the suite is already being changed so much.

Reply:
Done